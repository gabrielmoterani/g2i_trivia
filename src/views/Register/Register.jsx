// React Imports
import React, { useState, useEffect } from 'react'
import {StyleSheet, TouchableOpacity, View, Text, Image, TextInput, Button} from 'react-native';

// Libs Import
import { LinearGradient } from 'expo-linear-gradient';

// Components and Services
import Alert from '../../components/Alert';
import {saveRegisteredUser, checkIfStorageHasAnUser, removeRegisteredUser} from '../../services/asyncstorage';

//Static Imports
import LOGO from '../../assets/images/g2i.png'; 
import COLORS from '../../config/theme';


const Register = ({navigation}) => {

    const [name, setName] = useState('');

    const handleRegister = async () => {
        try{
            await saveRegisteredUser(name)
            redirectToWelcomePage()
        }
        catch (error) {
            Alert(error)
        } 
    }

    const redirectToWelcomePage = () => {
        navigation.replace('Welcome')
    }

    useEffect(() => {
        (async function checkIfItsRegistered() {
          const isRegisterd = await checkIfStorageHasAnUser();
          if(isRegisterd) redirectToWelcomePage()
        })();
    }, []);

    return ( 
        <LinearGradient style={styles.container} colors={COLORS.gradients[0]}>
            <View style={styles.registerSection}>
                <Text style={styles.label}>Welcome to Trivia Game</Text>
                <TextInput
                    style={styles.input}
                    value={name}
                    placeholder="Type your name to play!"
                    placeholderTextColor="#333"
                    onChangeText={text => setName(text)}
                ></TextInput>
                <TouchableOpacity 
                    onPress={() => handleRegister()}
                    style={styles.playBtn} 
                    disabled={name === ''}
                >
                    <Text style={styles.btnText}>Play!</Text>    
                </TouchableOpacity>
            </View>
        </LinearGradient>
     );
}

const styles = StyleSheet.create({
    container: {
      flex: 1,
      justifyContent: 'center',
      alignItems: 'center',
    },
    registerSection: {
        justifyContent: 'center',
        alignItems: 'center',
    },
    brandLogo:{
        marginBottom: 20
    },
    label:{
        color: COLORS.primary,
        fontSize: 20,
        fontWeight: '500',
        marginBottom: 20
    },
    input:{
        paddingVertical: 15,
        borderRadius: 5,
        backgroundColor: '#fff',
        alignSelf: 'stretch',
        textAlign: 'center',
        marginBottom: 15,
        marginBottom: 20
    },
    playBtn: {
        backgroundColor: COLORS.background,
        paddingVertical: 10,
        borderRadius: 5,
        alignSelf: 'stretch',
        alignItems: 'center'
    },
    btnText:{
        color: COLORS.primary,
        fontWeight: 'bold'
    }
  });


export default Register;