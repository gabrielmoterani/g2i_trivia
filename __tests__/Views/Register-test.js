import React from 'react';
import { shallow } from 'enzyme';
import Register from '../../src/views/Register';

const setup = () => shallow(<Register />);


describe('Test Register Screen', () => {
    const wrapper = setup();
  
    it('should render properly', () => {
      expect(wrapper).toMatchSnapshot();
    });
  
    it('should have a text Input', () => {
        const inputs = wrapper.find('TextInput');
        expect(inputs).toExist()
      }); 

    it('should have a button', () => {
      const btns = wrapper.find('TouchableOpacity');
      expect(btns).toExist()
    }); 
  
});