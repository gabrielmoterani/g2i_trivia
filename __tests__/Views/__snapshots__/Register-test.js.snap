// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Register Screen should render properly 1`] = `
<LinearGradient
  colors={
    Array [
      "#2C7EF8",
      "#2C7EF8",
      "#07ADD6",
    ]
  }
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 20,
          "fontWeight": "500",
          "marginBottom": 20,
        }
      }
    >
      Welcome to Trivia Game
    </Text>
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Type your name to play!"
      placeholderTextColor="#333"
      rejectResponderTermination={true}
      style={
        Object {
          "alignSelf": "stretch",
          "backgroundColor": "#fff",
          "borderRadius": 5,
          "marginBottom": 20,
          "paddingVertical": 15,
          "textAlign": "center",
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <TouchableOpacity
      activeOpacity={0.2}
      disabled={true}
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "backgroundColor": "#f00d00",
          "borderRadius": 5,
          "paddingVertical": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#fff",
            "fontWeight": "bold",
          }
        }
      >
        Play!
      </Text>
    </TouchableOpacity>
  </View>
</LinearGradient>
`;
